src/goto-symex/symex_function_call.cpp

```diff
@@ -86,24 +86,21 @@ void goto_symext::parameter_assignments(
       // It should be the same exact type.
       if(!base_type_eq(parameter_type, rhs.type(), ns))
       {
-        const typet &f_parameter_type=ns.follow(parameter_type);
-        const typet &f_rhs_type=ns.follow(rhs.type());
-
         // But we are willing to do some limited conversion.
         // This is highly dubious, obviously.
-        if((f_parameter_type.id()==ID_signedbv ||
-            f_parameter_type.id()==ID_unsignedbv ||
-            f_parameter_type.id()==ID_c_enum_tag ||
-            f_parameter_type.id()==ID_bool ||
-            f_parameter_type.id()==ID_pointer ||
-            f_parameter_type.id()==ID_union) &&
-           (f_rhs_type.id()==ID_signedbv ||
-            f_rhs_type.id()==ID_unsignedbv ||
-            f_rhs_type.id()==ID_c_bit_field ||
-            f_rhs_type.id()==ID_c_enum_tag ||
-            f_rhs_type.id()==ID_bool ||
-            f_rhs_type.id()==ID_pointer ||
-            f_rhs_type.id()==ID_union))
+        if(
```

Comment:
Would you mind using `// clang-format off` and keeping the one condition per line? I think this is one of the cases where it really makes a bad call.

Reply:
Indeed - fixed!